File: 1167-minimum-cost-to-connect-sticks/1167-minimum-cost-to-connect-sticks.py
class Solution:
    def connectSticks(self, sticks: List[int]) -> int:
        costs = 0
        heapq.heapify(sticks)
        
        while len(sticks) > 1:
            stick1 = heapq.heappop(sticks)
            stick2 = heapq.heappop(sticks)
            
            costs += (stick1 + stick2)
            heapq.heappush(sticks, stick1 + stick2)
        
        return costs
        

File: 1167-minimum-cost-to-connect-sticks/1167-minimum-cost-to-connect-sticks.java
class Solution {
    public int connectSticks(int[] sticks) {
        int minCost = 0;
        
        PriorityQueue<Integer> minHeap = new PriorityQueue<Integer>((x,y) -> Integer.compare(x,y));
        
        for (int stick: sticks) {
            minHeap.add(stick);
        }
        
        for (int i = minHeap.size(); i > 1; i--) {
            int v1 = minHeap.poll(), v2 = minHeap.poll();
            int newStick = v1+v2;
            minHeap.add(newStick);
            
            minCost += newStick;
        }
        
        return minCost;
    }
}